src/Symfony/Component/Routing/RouteCollectionBuilder.php

```diff
@@ -311,6 +311,9 @@ public function build()
                 $subCollection->addPrefix($this->prefix);
 
                 $routeCollection->addCollection($subCollection);
+                foreach ($route->resources as $resource) {
```

Comment:
shouldn't this be handled by `$routeCollection->addCollection` instead ?

Reply:
This doesn't work because `$route->build()` (https://github.com/symfony/symfony/pull/24642/files#diff-61a7c46324774b76b342efb3f3e87b7fL310) doesn't add sub-resources to `$subCollection`. This is exactly what this patch fix.

Reply:
it is adding them just a few lines below in this code.

Reply:
`$route->resources` vs `$this->resources`

Reply:
While you are on this file... it looks like nothing calls the private method `addResource`, so maybe you can remove the `resources` property of that class, as well as the loop "below".

Reply:
sure. But the loop below happens inside `->build()` and so the subcollection **will** have the resources in it if it has any.

Reply:
But it’s not the case (the new test fail without my patch). I’ll Investigate this afternoon.

Reply:
I double checked and pushed a better fix:

`$route` can have resources but no routes. So the loop over `$this->resources` must be outside the loop on `$this->routes`.

Can you check if it's ok @stof?